src/GraphQL/Types/Scalars/FloatGraphType.cs

```diff
@@ -10,30 +10,49 @@ namespace GraphQL.Types
     public class FloatGraphType : ScalarGraphType
     {
         /// <inheritdoc/>
-        public override object? ParseLiteral(IValue value) => value switch
+        public override object? ParseLiteral(GraphQLValue value)
```

Comment:
@Shane32 I need help here with design. Before changes we had different nodes for each clr type where each node handled its own value type. Now there are less node types and in case of `IntValueNode` and `FloatValueNode` node's value now may be int, long, decimal, etc. This is not such a big problem, but all that parsing refactoring is very tedious.

Reply:
This one is an already migrated example. Other scalars are not yet compiled.

Reply:
How about that ?
```c#
 public override object? ParseLiteral(GraphQLValue value)
            => value switch
            {
                GraphQLIntValue x => Double.Parse(x.Value),
                GraphQLFloatValue x => Double.Parse(x.Value),
                GraphQLNullValue _ => null,
                _ => ThrowLiteralConversionError(value)
            };

        /// <inheritdoc/>
        public override bool CanParseLiteral(GraphQLValue value)
            => value switch
            {
                GraphQLIntValue x => Double.TryParse(x.Value, out double _),
                GraphQLFloatValue x => Double.TryParse(x.Value, out double _),
                GraphQLNullValue _ => true,
                _ => false
            };
```

Double is a shim:
```c#
 internal static class Double
    {
#if NETSTANDARD2_1
        public static bool TryParse(ReadOnlySpan<char> s, out double result)
           => double.TryParse(s, NumberStyles.AllowLeadingSign | NumberStyles.AllowDecimalPoint | NumberStyles.AllowExponent, CultureInfo.InvariantCulture, out result);
        public static bool TryParse(ReadOnlySpan<char> s, NumberStyles style, IFormatProvider provider, out double result)
            => double.TryParse(s, style, provider, out result);
        public static double Parse(ReadOnlySpan<char> s, NumberStyles style, IFormatProvider provider)
            => double.Parse(s, style, provider);
        public static double Parse(ReadOnlySpan<char> s)
            => double.Parse(s, NumberStyles.AllowLeadingSign | NumberStyles.AllowDecimalPoint | NumberStyles.AllowExponent, CultureInfo.InvariantCulture);
...

So there should be no any complicated code in all numeric scalars - parsing all combinations of numeric conversions. We can use built-in system `TryParse/Parse` methods for numbers - double/int/long/uint.....etc . Worth to think about.  it It is easier for support. 
```

Reply:
If we did such a change then what is the point of caching the value in the parser?

Reply:
I like the idea but we should remove the value caching in the parser.

Reply:
Realistically the scalar values are only read by the scalars and only once. So caching is typically not a benefit and would only consume more memory.

Reply:
> If we did such a change then what is the point of caching the value in the parser?

Exactly. When I wrote this example, I immediately thought about it and decided to check. After refactoring all the scalars it became obvious that everything works without touching `ClrValue`.

> Realistically the scalar values are only read by the scalars and only once. So caching is typically not a benefit and would only consume more memory.

Yes!

Reply:
Done.